collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -148,6 +150,19 @@ public static String getMd5Hash(File file) {
         return getMd5Hash(is);
     }
 
+    public static String getMd5Hash(String message) throws NoSuchAlgorithmException {
```

Comment:
This method will shorten to:
```
    public static String getMd5Hash(String message) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("MD5");
        md.update(message.getBytes());
        StringBuilder builder = new StringBuilder();
        for (byte b : md.digest()) {
            builder.append(String.format("%02x", b));
        }
        return builder.toString();
    }
```

But instead I suggest you consider using the new DigestAlgorithm we just added to JavaRosa.

Reply:
Thanks for letting us know! I'll try the `DigestAlgorithm` from `Javarosa`.
Btw, I also really liked the reduced `getMd5Hash()` function.

Reply:
The `Encoding` enum in javarosa is not public. So, I cannot use it directly for encoding. I tried out the below line use the javarosa's algorithm. 

```
String md5Hash = DigestAlgorithm.MD5.digest(message, Encoding.BASE64);
```

Reply:
I’ve had second thoughts about using JavaRosa for that. I no longer think JavaRosa should share that.

Reply:
Yeah I saw your comment on #general channel and currently I'm trying out the apache commons lib

Reply:
```
    public static String getMd5Hash(String message) {
        DigestUtils digestUtils = new DigestUtils("MD5");
        digestUtils.digest(message);
        StringBuilder builder = new StringBuilder();
        for (byte b : digestUtils.digest(message)) {
            builder.append(String.format("%02x", b));
        }
        return builder.toString();
    }
```

Reply:
Doesn’t it have a method to do the hex encoding, too?

Reply:
Or even just `DigestUtils.md5Hex(message)`

Reply:
That’s what I was thinking of.